api.js

```diff
@@ -770,25 +783,46 @@ module.exports = async function attachAPI(app, {wss, db}) {
           return
         }
 
-        connectedSocketsMap.set(socket, {
+        Object.assign(connectedSocketsMap.get(socket), {
           channelID: data // channelID
         })
+      } else if (evt === 'pong data') {
+        // Not the built-in pong; this event is used for gathering
+        // socket-specific data.
```

Comment:
Hey, uh, `socket.ping` can take a data argument. Check out the ws documentation.

Reply:
I didn't see a way to detect the actual ping event from the browser, though? https://stackoverflow.com/a/45929287/4633828

Reply:
Ok!